src/storage/storage.service.ts

```diff
@@ -0,0 +1,72 @@
+import {
+    Injectable,
+    InternalServerErrorException,
+    Logger,
+    LoggerService,
+} from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import {
+    ReadStream,
+    WriteStream,
+    createReadStream,
+    createWriteStream,
+} from 'fs';
+import { readdir } from 'fs/promises';
+import { join } from 'path';
+import { v4 as uuid } from 'uuid';
+import * as fs from 'fs';
+
+@Injectable()
+export class StorageService {
+    constructor(private readonly configService: ConfigService) {}
+
+    async saveFile(file: Express.Multer.File): Promise<string> {
+        const storage: string | undefined =
+            this.configService.get<string>('STORAGE');
+
+        if (!storage) {
+            throw new Error('Something wrong with env or config service');
+        }
+
+        const fileId: string = uuid();
+        const fileExtension: string = file.mimetype.includes('html')
+            ? 'html'
+            : 'xml';
+        const writeStream: WriteStream = createWriteStream(
+            join(process.cwd(), `${storage}/${fileId}.${fileExtension}`),
+        );
+        writeStream.write(file);
+
+        return fileId;
```

Comment:
"The writable.write() method writes some data to the stream, and calls the supplied callback once the data has been fully handled."
Jak zwracamy fileId zanim dane są zapisane, to może się w skrajnych wypadkach gdzieś wywalić

Reply:
Fakt, tutaj można użyć jakiś Promise do zapewnienia, że plik został zapisany. Aktualnie może wyskoczyć błąd a fileId zostanie zwrócone. Do poprawki.

Reply:
czyli zostaje tak jak jest